src/main/groovy/com/tacitknowledge/gradle/sass/SassWatchTask.groovy

```diff
@@ -1,20 +1,87 @@
 package com.tacitknowledge.gradle.sass
 
+import java.nio.file.FileSystems
+import java.nio.file.FileVisitResult
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.nio.file.SimpleFileVisitor
+import java.nio.file.StandardCopyOption
+import java.nio.file.StandardWatchEventKinds
+import java.nio.file.WatchKey
+import java.nio.file.WatchService
+import java.nio.file.attribute.BasicFileAttributes
+
 class SassWatchTask extends SassTask
 {
-  SassWatchTask()
-  {
+
+  WatchService watcher
+  Map<WatchKey, Path> keys
+
+  SassWatchTask() {
     ignoreExitValue = true
+    this.watcher = FileSystems.getDefault().newWatchService();
+    this.keys = new HashMap<WatchKey, Path>();
+  }
+
+  void registerRecursive(Path start) throws IOException {
+    Files.walkFileTree(start, new SimpleFileVisitor<Path>() {
+      @Override
+      FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
+        registerDir(dir);
+        return FileVisitResult.CONTINUE;
+      }
+    });
+  }
+
+  private void registerDir(Path path) {
+    println "Registering  $path to watch for changes"
+    WatchKey key = path.register(this.watcher,
+            StandardWatchEventKinds.ENTRY_CREATE,
+            StandardWatchEventKinds.ENTRY_DELETE,
+            StandardWatchEventKinds.ENTRY_MODIFY);
+    keys.put(key, path);
   }
 
   @Override
-  void exec()
-  {
-    Thread.start {
+  void exec() {
+
+    project.sass.watchList.each { String dir ->
+      Path path = Paths.get(dir)
+      registerRecursive(path)
+
+      Thread.start {
```

Comment:
There are as many threads started as there are entries in watchList. Is this expected?